tests/components/project/test_views.py

```diff
@@ -303,10 +331,23 @@ async def test_upload_project_logo_calls_put_object_method_and_updates_project_l
         assert s3_test_client.check_if_bucket_exists(settings.S3_BUCKET_FOR_PROJECT_LOGOS)
         assert s3_test_client.check_if_file_exists(settings.S3_BUCKET_FOR_PROJECT_LOGOS, expected_logo_name)
 
-    async def test_upload_project_logo_returns_not_found_response_for_non_existing_project(self, client, fake):
+    async def test_upload_project_logo_returns_not_found_response_for_non_existing_project(
+        self, client, fake, minio_container
+    ):
         project_id = fake.uuid4()
         payload = {'base64': fake.base64_image()}
 
         response = await client.post(f'/v1/projects/{project_id}/logo', json=payload)
 
         assert response.status_code == 404
+
+    async def test_upload_project_logo_returns_error_for_too_big_image(
+        self, project_factory, client, fake, minio_container
+    ):
+        project = await project_factory.create()
+
+        payload = {'base64': 'a' * 2**30}
```

Comment:
I would suggest to use here a value from the settings as a boundary check.
`2**30` value will take more that 1gb of memory for this test.
```suggestion
        payload = {'base64': 'a' * (settings.ICON_SIZE_LIMIT + 1)}
```